GridMvc/Columns/GridColumn.cs

```diff
@@ -194,6 +196,22 @@ public override IGridCell GetValue(T instance)
                     textValue = string.Empty;
                 else if (!string.IsNullOrEmpty(ValuePattern))
                     textValue = string.Format(ValuePattern, value);
+                else if (typeof (TDataType).IsEnum)
+                {
+                    try
+                    {
+                        var attributes = typeof (TDataType).GetMember(value.ToString())
+                            .First()
+                            .GetCustomAttributes(false);
+
+                        textValue = attributes.OfType<DisplayAttribute>().First().Name;

```

Comment:
Sure, feel free to change it. I come from python background so I like to catch exceptions :)
14 kwi 2015 11:04 PM Alexander Zaytsev <notifications@github.com> napisał(a):In GridMvc/Columns/GridColumn.cs:
> @@ -194,6 +196,22 @@ public override IGridCell GetValue(T instance)
>                      textValue = string.Empty;
>                  else if (!string.IsNullOrEmpty(ValuePattern))
>                      textValue = string.Format(ValuePattern, value);
> +                else if (typeof (TDataType).IsEnum)
> +                {
> +                    try
> +                    {
> +                        var attributes = typeof (TDataType).GetMember(value.ToString())
> +                            .First()
> +                            .GetCustomAttributes(false);
> +
> +                        textValue = attributes.OfType<DisplayAttribute>().First().Name;

Should we check that Attributes actually present instead of catching exception?

—Reply to this email directly or view it on GitHub.